#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.2907054723834543 -0.8655154424591766 -0.40788889074233003 2.0361447836683655
  position -4.092078001481153 4.699428369704343 -1.042188345489295
  follow "robot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 5 5
}
Robot {
  translation -1 0.045 0
  rotation 0 1 0 3.09
  children [
    DEF BODY_GR Group {
      children [
        Transform {
          translation 0 0 0.06495
          rotation 0 1 0 0.523583333
          children [
            Shape {
              appearance DEF COLOR_BODY_ROBOT PBRAppearance {
                baseColor 0.7294117647058823 0.7411764705882353 0.7137254901960784
              }
              geometry DEF GEOMETRY_BODY_ROBOT Box {
                size 0.25 0.05 0.005
              }
            }
          ]
        }
        Transform {
          translation 0 0 -0.06495
          rotation 0 1 0 -0.523583
          children [
            Shape {
              appearance USE COLOR_BODY_ROBOT
              geometry USE GEOMETRY_BODY_ROBOT
            }
          ]
        }
        Transform {
          translation -0.111 0 0
          rotation 0 1 0 1.57
          children [
            Shape {
              appearance USE COLOR_BODY_ROBOT
              geometry USE GEOMETRY_BODY_ROBOT
            }
          ]
        }
        Transform {
          translation -0.11 0.025 -0.00069
          rotation 0 1 0 3.921592644
          children [
            Shape {
              appearance USE COLOR_BODY_ROBOT
              geometry IndexedFaceSet {
                coord Coordinate {
                  point [
                    0.09 0 -0.09
                    -0.155 0 -0.155
                    -0.09 0 0.09
                  ]
                }
                solid FALSE
                convex FALSE
                coordIndex [
                  0, 1, 2, -1
                ]
              }
            }
          ]
        }
      ]
    }
    DEF WHEELS_GR Group {
      children [
        DEF WHEEL_1 Transform {
          translation -0.115 0 -0.136
          rotation 0 1 0 0.52
          children [
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
              }
              device [
                RotationalMotor {
                  name "motor1"
                  maxVelocity 30.36872
                  maxTorque 0.847386219999996
                }
                PositionSensor {
                  name "position_sensor1"
                  resolution 0.003848528
                }
              ]
              endPoint Solid {
                translation 3.850324754901607e-05 -7.119440102921257e-05 -1.1014670775466584e-06
                rotation -2.0303258352144598e-05 -3.4708346730909494e-05 -0.9999999991915542 0.2712929498997691
                children [
                  DEF WHEELS_TR1 Transform {
                    rotation 1 0 0 1.57
                    children [
                      DEF WHEELS_SH Shape {
                        appearance DEF COLOR_WHEEL PBRAppearance {
                          baseColor 0 0 0
                          roughness 1
                          metalness 0
                        }
                        geometry DEF SIZE_WHEEL Cylinder {
                          height 0.02
                          radius 0.04
                        }
                      }
                    ]
                  }
                  DEF SW1_1 HingeJoint {
                    jointParameters HingeJointParameters {
                      anchor 0 0.04 0
                    }
                    endPoint Solid {
                      translation 0 0.04 0
                      rotation 0 0 1 1.57
                      children [
                        DEF SMALL_WHEEL__TR1 Transform {
                          children [
                            DEF SMALL_WHEEL_SH Shape {
                              appearance PBRAppearance {
                                roughness 1
                                metalness 0
                              }
                              geometry Cylinder {
                                height 0.02
                                radius 0.005
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE SMALL_WHEEL__TR1
                      physics Physics {
                      }
                    }
                  }
                  DEF SW1_2 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 -1 0
                      anchor 0.02828 0.02828 0
                    }
                    endPoint Solid {
                      translation 0.02828 0.02828 0
                      rotation 0 0 1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(1)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW1_3 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 1 0
                      anchor -0.02828 0.02828 0
                    }
                    endPoint Solid {
                      translation -0.02828 0.02828 0
                      rotation 0 0 -1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(2)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW1_4 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 1 0
                      anchor -0.04 0 0
                    }
                    endPoint Solid {
                      translation -0.04 0 0
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(3)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW1_5 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 1 0
                      anchor 0.04 0 0
                    }
                    endPoint Solid {
                      translation 0.04 0 0
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(4)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW1_6 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 -1 0
                      anchor -0.02828 -0.02828 0
                    }
                    endPoint Solid {
                      translation -0.02828 -0.02828 0
                      rotation 0 0 1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(5)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW1_7 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 1 0
                      anchor 0.02828 -0.02828 0
                    }
                    endPoint Solid {
                      translation 0.02828 -0.02828 0
                      rotation 0 0 -1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(6)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW1_8 HingeJoint {
                    jointParameters HingeJointParameters {
                      anchor 0 -0.04 0
                    }
                    endPoint Solid {
                      translation 0 -0.04 0
                      rotation 0 0 1 1.57
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(7)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                ]
                name "solid(2)"
                boundingObject USE WHEELS_TR1
                physics Physics {
                  density 2500
                }
              }
            }
          ]
        }
        DEF WHEEL_2 Transform {
          translation -0.115 0 0.136
          rotation 0 1 0 -0.52
          children [
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
              }
              device [
                RotationalMotor {
                  name "motor2"
                  maxVelocity 30.36872
                  maxTorque 0.847386219999996
                }
                PositionSensor {
                  name "position_sensor2"
                  resolution 0.003848528
                }
              ]
              endPoint Solid {
                translation 9.007236958713253e-05 7.199735075744964e-05 7.539973861936389e-07
                rotation 0.00020818493322224466 -0.0001456954946994768 -0.9999999677159277 0.0739737941855639
                children [
                  DEF WHEEL_TR2 Transform {
                    rotation 1 0 0 1.57
                    children [
                      USE WHEELS_SH
                    ]
                  }
                  DEF SW2_1 HingeJoint {
                    jointParameters HingeJointParameters {
                      anchor 0 0.04 0
                    }
                    endPoint Solid {
                      translation 0 0.04 0
                      rotation 0 0 1 1.57
                      children [
                        DEF SMALL_WHEEL_TR2 Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      boundingObject USE SMALL_WHEEL_TR2
                      physics Physics {
                      }
                    }
                  }
                  DEF SW2_2 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 -1 0
                      anchor 0.02828 0.02828 0
                    }
                    endPoint Solid {
                      translation 0.02828 0.02828 0
                      rotation 0 0 1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(1)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW2_3 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 1 0
                      anchor -0.02828 0.02828 0
                    }
                    endPoint Solid {
                      translation -0.02828 0.02828 0
                      rotation 0 0 -1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(2)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW2_4 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 1 0
                      anchor -0.04 0 0
                    }
                    endPoint Solid {
                      translation -0.04 0 0
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(3)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW2_5 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 1 0
                      anchor 0.04 0 0
                    }
                    endPoint Solid {
                      translation 0.04 0 0
                      rotation 0 1 0 1.57
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(4)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW2_6 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 -1 0
                      anchor -0.02828 -0.02828 0
                    }
                    endPoint Solid {
                      translation -0.02828 -0.02828 0
                      rotation 0 0 1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(5)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW2_7 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 1 0
                      anchor 0.02828 -0.02828 0
                    }
                    endPoint Solid {
                      translation 0.02828 -0.02828 0
                      rotation 0 0 -1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(6)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW2_8 HingeJoint {
                    jointParameters HingeJointParameters {
                      anchor 0 -0.04 0
                    }
                    endPoint Solid {
                      translation 0 -0.04 0
                      rotation 0 0 1 1.57
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(7)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                ]
                boundingObject USE WHEEL_TR2
                physics Physics {
                  density 2500
                }
              }
            }
          ]
        }
        DEF WHEEL_3 Transform {
          translation 0.115 0 0
          rotation 0 1 0 1.57
          children [
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
              }
              device [
                RotationalMotor {
                  name "motor3"
                  maxVelocity 30.36872
                  maxTorque 0.847386219999996
                }
                PositionSensor {
                  name "position_sensor3"
                  resolution 0.003848528
                }
              ]
              endPoint Solid {
                translation -2.5891360131228366e-05 0.00017763253641808345 1.6115172847275258e-05
                rotation 4.376412593902718e-06 -9.938050607107914e-07 0.9999999999899296 6.0969936636291875
                children [
                  DEF WHEEL_TR3 Transform {
                    rotation 1 0 0 1.57
                    children [
                      USE WHEELS_SH
                    ]
                  }
                  DEF SW3_1 HingeJoint {
                    jointParameters HingeJointParameters {
                      anchor 0 0.04 0
                    }
                    endPoint Solid {
                      translation 0 0.04 0
                      rotation 0 0 1 1.57
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW3_2 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 -1 0
                      anchor 0.02828 0.02828 0
                    }
                    endPoint Solid {
                      translation 0.02828 0.02828 0
                      rotation 0 0 1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(1)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW3_3 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 1 0
                      anchor -0.02828 0.02828 0
                    }
                    endPoint Solid {
                      translation -0.02828 0.02828 0
                      rotation 0 0 -1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(2)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW3_4 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 1 0
                      anchor -0.04 0 0
                    }
                    endPoint Solid {
                      translation -0.04 0 0
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(3)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW3_5 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 1 0
                      anchor 0.04 0 0
                    }
                    endPoint Solid {
                      translation 0.04 0 0
                      rotation 0 1 0 1.57
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(4)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW3_6 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 -1 0
                      anchor -0.02828 -0.02828 0
                    }
                    endPoint Solid {
                      translation -0.02828 -0.02828 0
                      rotation 0 0 1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(5)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW3_7 HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 1 1 0
                      anchor 0.02828 -0.02828 0
                    }
                    endPoint Solid {
                      translation 0.02828 -0.02828 0
                      rotation 0 0 -1 0.7852999999999999
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(6)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                  DEF SW3_8 HingeJoint {
                    jointParameters HingeJointParameters {
                      anchor 0 -0.04 0
                    }
                    endPoint Solid {
                      translation 0 -0.04 0
                      rotation 0 0 1 1.57
                      children [
                        Transform {
                          children [
                            USE SMALL_WHEEL_SH
                          ]
                        }
                      ]
                      name "solid(7)"
                      boundingObject USE SMALL_WHEEL_SH
                      physics Physics {
                      }
                    }
                  }
                ]
                name "solid(1)"
                boundingObject USE WHEEL_TR3
                physics Physics {
                  density 2500
                }
              }
            }
          ]
        }
      ]
    }
    DEF DISTANCE_SENSOR_RIGHT DistanceSensor {
      translation -0.115 0 -0.05
      rotation 0 1 0 3.14159
      children [
        Transform {
          rotation 0 1 0 -0.0872665
          children [
            DEF DISTANCE_SENSOR_SH Shape {
              appearance PBRAppearance {
                baseColor 0.9882352941176471 0.9137254901960784 0.30980392156862746
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
        }
      ]
      name "distance_sensor1"
      lookupTable [
        0 0 0
        0.2 65535 0
      ]
      resolution 1
    }
    DEF DISTANCE_SENSOR_RIGHT DistanceSensor {
      translation -0.115 0 0.05
      rotation 0 1 0 3.14159
      children [
        Transform {
          rotation 0 1 0 0.0872665
          children [
            USE DISTANCE_SENSOR_SH
          ]
        }
      ]
      name "distance_sensor2"
      lookupTable [
        0 0 0
        0.2 65535 0
      ]
    }
  ]
  boundingObject USE BODY_GR
  physics Physics {
    density 4500
    centerOfMass [
      0 0 0
    ]
  }
  controller "u3_project_Espinoza_Angelo"
}
Solid {
  translation 0 0.25 -2.5
  children [
    DEF WALLS_SH Shape {
      appearance PBRAppearance {
        baseColor 0.5607843137254902 0.34901960784313724 0.00784313725490196
        roughness 1
        metalness 0
      }
      geometry Box {
        size 5 0.5 0.001
      }
    }
  ]
  boundingObject USE WALLS_SH
}
Solid {
  translation -2.5 0.25 0
  rotation 0 1 0 1.57
  children [
    USE WALLS_SH
  ]
  name "solid(1)"
  boundingObject USE WALLS_SH
}
Solid {
  translation 0 0.25 2.5
  children [
    USE WALLS_SH
  ]
  name "solid(2)"
  boundingObject USE WALLS_SH
}
Solid {
  translation 2.5 0.25 0
  rotation 0 1 0 1.57
  children [
    USE WALLS_SH
  ]
  name "solid(3)"
  boundingObject USE WALLS_SH
}
Solid {
  translation 0 0.25 -0.14
  children [
    DEF OBSTACLE1_SH Shape {
      appearance PBRAppearance {
        baseColor 0.8 0 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(4)"
  boundingObject USE OBSTACLE1_SH
}
Solid {
  translation -0.3728151191443926 0.1 0.7226715533280423
  rotation 0 -1 0 0.0022128730000013194
  children [
    DEF OBSTACLE2_SH Shape {
      appearance PBRAppearance {
        baseColor 0.9882352941176471 0.9137254901960784 0.30980392156862746
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.2 0.2
      }
    }
  ]
  name "solid(5)"
  boundingObject USE OBSTACLE2_SH
}
Solid {
  translation 1.2400000000000007 0.25 -0.12000000000000001
  children [
    DEF OBSTACLE3_SH Shape {
      appearance PBRAppearance {
        baseColor 0.12549019607843137 0.2901960784313726 0.5294117647058824
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(6)"
  boundingObject USE OBSTACLE3_SH
}
Solid {
  translation 1.670000000000001 0.35 0.5600000000000002
  children [
    DEF OBSTACLE4_SH Shape {
      appearance PBRAppearance {
        baseColor 0.3058823529411765 0.6039215686274509 0.023529411764705882
        roughness 1
        metalness 0
      }
      geometry Capsule {
        height 0.5
        radius 0.09
      }
    }
  ]
  name "solid(7)"
  boundingObject USE OBSTACLE4_SH
}
